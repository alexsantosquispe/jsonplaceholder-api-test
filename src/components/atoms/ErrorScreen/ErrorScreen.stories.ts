import { Meta, StoryObj } from '@storybook/react';

import { ErrorScreen } from './ErrorScreen';

const meta = {
  title: 'Atoms/ErrorScreen',
  component: ErrorScreen,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof ErrorScreen>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {}
};
