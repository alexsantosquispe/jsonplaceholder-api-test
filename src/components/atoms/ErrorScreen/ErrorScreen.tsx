import { twMerge } from 'tailwind-merge';

import { ExclamationOutlineIcon } from '../../../icons';

interface ErrorScreenProps {
  message?: string;
  className?: string;
}

const defaultMessage = 'Ups! Something went wrong. Please try again later.';

export const ErrorScreen = ({ message, className }: ErrorScreenProps) => {
  return (
    <div
      className={twMerge(
        'flex w-full items-center justify-center px-4',
        className
      )}
    >
      <div className="flex flex-col items-center justify-center gap-y-4 rounded-xl border border-gray-200 p-8 shadow-xs md:p-40 dark:border-white/10">
        <ExclamationOutlineIcon className="size-8 text-red-600 md:size-10" />
        <span className="text-primary text-center font-medium md:text-2xl dark:text-white">
          {message || defaultMessage}
        </span>
      </div>
    </div>
  );
};
