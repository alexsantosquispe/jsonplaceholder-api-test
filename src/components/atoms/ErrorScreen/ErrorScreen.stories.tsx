import { Meta, StoryObj } from '@storybook/react';

import { ErrorScreen } from './ErrorScreen';
import { Wrapper } from '../../../utils/stories-utils';

const meta = {
  title: 'Atoms/ErrorScreen',
  component: ErrorScreen,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof ErrorScreen>;

export default meta;

type Story = StoryObj<typeof ErrorScreen>;

export const Default: Story = () => {
  return (
    <Wrapper>
      <ErrorScreen />
    </Wrapper>
  );
};

Default.args = {};
