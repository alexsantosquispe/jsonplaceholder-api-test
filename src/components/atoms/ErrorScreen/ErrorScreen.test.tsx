import { render } from '@testing-library/react';

import { ErrorScreen } from './ErrorScreen';

describe('ErrorScreen', () => {
  it('should render the component correctly', () => {
    const component = render(<ErrorScreen />);

    expect(component).toMatchSnapshot();
  });
});
