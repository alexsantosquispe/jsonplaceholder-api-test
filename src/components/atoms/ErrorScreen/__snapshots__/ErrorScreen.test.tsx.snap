// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="-mt-[4rem] flex w-full items-center justify-center px-4"
      >
        <div
          class="flex flex-col items-center justify-center gap-y-4 rounded-xl border border-gray-200 p-8 shadow-xs md:p-40 dark:border-white/10"
        >
          <svg
            class="size-8 text-red-600 md:size-10"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 9v3.75m9-.75a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9 3.75h.008v.008H12v-.008Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <span
            class="text-primary text-center font-medium md:text-2xl dark:text-white"
          >
            Ups! Something went wrong. Please try again later.
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="-mt-[4rem] flex w-full items-center justify-center px-4"
    >
      <div
        class="flex flex-col items-center justify-center gap-y-4 rounded-xl border border-gray-200 p-8 shadow-xs md:p-40 dark:border-white/10"
      >
        <svg
          class="size-8 text-red-600 md:size-10"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 9v3.75m9-.75a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9 3.75h.008v.008H12v-.008Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="text-primary text-center font-medium md:text-2xl dark:text-white"
        >
          Ups! Something went wrong. Please try again later.
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
